infra/shared/data-scripts/user-data-server.sh

```diff
@@ -86,3 +86,91 @@ consul kv put -token-file=$CONSUL_BOOTSTRAP_TOKEN nomad_user_token "$(cat $NOMAD
 
 echo "ACL bootstrap end"
 
+sudo systemctl restart vault
+
+sleep 10
+
+export VAULT_ADDR=http://127.0.0.1:8200
+echo "export VAULT_ADDR=http://127.0.0.1:8200" >> ~/.bashrc
+
+touch /etc/vault.d/init.file
+vault operator init > /etc/vault.d/init.file
+
+echo -e '\e[38;5;198m'"++++ Auto unseal vault"
+
+for i in $(cat /etc/vault.d/init.file | grep Unseal | cut -d " " -f4 | head -n 3); do vault operator unseal $i; done
+VAULT_TOKEN=$(grep 'Initial Root Token' /etc/vault.d/init.file | cut -d ':' -f2 | tr -d ' ')
+
+consul kv put -token-file=$CONSUL_BOOTSTRAP_TOKEN vault_token "$VAULT_TOKEN"
+
+tee nomad-server-policy.hcl <<EOF
+
+# Allow creating tokens under "nomad-cluster" token role. The token role name
+# should be updated if "nomad-cluster" is not used.
+path "auth/token/create/nomad-cluster" {
+  capabilities = ["update"]
+}
+
+# Allow looking up "nomad-cluster" token role. The token role name should be
+# updated if "nomad-cluster" is not used.
+path "auth/token/roles/nomad-cluster" {
+  capabilities = ["read"]
+}
+
+# Allow looking up the token passed to Nomad to validate # the token has the
+# proper capabilities. This is provided by the "default" policy.
+path "auth/token/lookup-self" {
+  capabilities = ["read"]
+}
+
+# Allow looking up incoming tokens to validate they have permissions to access
+# the tokens they are requesting. This is only required if
+# `allow_unauthenticated` is set to false.
+path "auth/token/lookup" {
+  capabilities = ["update"]
+}
+
+# Allow revoking tokens that should no longer exist. This allows revoking
+# tokens for dead tasks.
+path "auth/token/revoke-accessor" {
+  capabilities = ["update"]
+}
+
+# Allow checking the capabilities of our own token. This is used to validate the
+# token upon startup.
+path "sys/capabilities-self" {
+  capabilities = ["update"]
+}
+
+# Allow our own token to be renewed.
+path "auth/token/renew-self" {
+  capabilities = ["update"]
+}
+EOF
+
+vault login $VAULT_TOKEN
+
+vault policy write nomad-server nomad-server-policy.hcl
+
+tee nomad-cluster-role.json <<EOF
+{
+  "allowed_policies": "access-tables",
+  "token_explicit_max_ttl": 0,
+  "name": "nomad-cluster",
+  "orphan": true,
+  "token_period": 259200,
+  "renewable": true
+}
+EOF
+
+vault write /auth/token/roles/nomad-cluster @nomad-cluster-role.json
+
+tee access-tables-policy.hcl <<EOF
+path "database/creds/accessdb" {
+  capabilities = ["read"]
+}
+EOF
+
+vault policy write access-tables access-tables-policy.hcl
+
+vault token create -policy nomad-server -period 72h -orphan
```

Comment:
This token shall be consumed/configured by nomad server. Will this be done in next PR?